Tests/scripts/download_conf_repos.sh

```diff
@@ -1,11 +1,53 @@
 #!/usr/bin/env bash
 set +e
 
-# replace slashes ('/') in the branch name, if exist, with underscores ('_')
+clone_repository() {
+  local repo_name=$1
+  local branch=$2
+  local retry_count=$3
+  local exit_code=0
+  local i=1
+  echo "clone ${repo_name} from branch: ${branch} with ${retry_count} retries"
+  for ((i=1; i <= retry_count; i++)); do
+    git clone --depth=1 "https://gitlab-ci-token:${CI_JOB_TOKEN}@${repo_name}.git" --branch "${branch}" && exit_code=0 && break || exit_code=$?
+    if [ ${i} -ne "${retry_count}" ]; then
+      echo "Failed to clone ${repo_name} with branch:${branch}, exit code:${exit_code}, sleeping for 10 seconds and trying again"
+      sleep 10
```

Comment:
Does it make sense to keep the `10` in a local variable, like `i`?
So it's easy to change.